x-pack/legacy/plugins/ml/public/services/http_service.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// service for interacting with the server
+
+import chrome from 'ui/chrome';
+
+// @ts-ignore
+import { addSystemApiHeader } from 'ui/system_api';
+import { fromFetch } from 'rxjs/fetch';
+import { from, Observable } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+
+export interface HttpOptions {
+  url?: string;
+}
+
+function getResultHeaders(headers: HeadersInit): HeadersInit {
+  return addSystemApiHeader({
+    'Content-Type': 'application/json',
+    'kbn-version': chrome.getXsrfToken(),
+    ...headers,
+  });
+}
+
+export function http(options: any) {
+  return new Promise((resolve, reject) => {
+    if (options && options.url) {
+      let url = '';
+      url = url + (options.url || '');
+      const headers: Record<string, string> = addSystemApiHeader({
+        'Content-Type': 'application/json',
+        'kbn-version': chrome.getXsrfToken(),
+        ...options.headers,
+      });
+
+      const allHeaders =
+        options.headers === undefined ? headers : { ...options.headers, ...headers };
+      const body = options.data === undefined ? null : JSON.stringify(options.data);
+
+      const payload: RequestInit = {
+        method: options.method || 'GET',
+        headers: allHeaders,
+        credentials: 'same-origin',
+      };
+
+      if (body !== null) {
+        payload.body = body;
+      }
+
+      fetch(url, payload)
+        .then(resp => {
+          resp.json().then(resp.ok === true ? resolve : reject);
+        })
+        .catch(resp => {
+          reject(resp);
+        });
+    } else {
+      reject();
+    }
+  });
+}
+
+export function httpCall<T>(url: string, options: RequestInit): Observable<T> {
```

Comment:
Would `http$` instead of `httpCall` be an alternative name we could use to be in line with other observable's `$` postfix?

Reply:
Usually, postfix `$` is used for variables that contain Observable or Subject, but not for methods that return them. But I'm fine with renaming it, let's wait for other suggestions and opinions 🙂

Reply:
done in https://github.com/elastic/kibana/pull/51008/commits/ce4825c6bfa54eb7500e2690ad8ef293431d4f5a